src/lib/utils.py

```diff
@@ -386,3 +386,33 @@ def derive_localities(localities: DataFrame, data: DataFrame) -> DataFrame:
     locs["key"] = locs["locality"]
     index_columns = ["key", "date"] if "date" in data.columns else ["key"]
     return locs.groupby(index_columns).agg(agg_func)
+
+
+def backfill_cumulative_fields(data: DataFrame, columns=[]) -> DataFrame:
```

Comment:
Don't use a list for a default parameter value in Python: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Used `None` instead.